### PR TITLE
[WIP] Featured bounties

### PR DESCRIPTION
**Pull request by octavioamu**
##### Checklist

- [ ] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [ ] Linter status: 100% pass
- [ ] Added relevant tests
- [ ] Tests pass and code coverage has not decreased
- [ ] Changes don't break existing behavior
- [ ] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [ ] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [ ] Completed: Code review by core team and any requested changes

##### Refers/Fixes

Refs: #2390 
Related : #3508 #3517 #3510 

###### Contributor

- [ ] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Tested all changes **locally**
- [ ] Verified existing functionality
- [ ] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging or local deployment

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

File: app/assets/v2/js/pages/new_bounty.js [hunk 1]
```diff
@@ -440,7 +442,31 @@ $(document).ready(function() {
         ipfs.addJson(ipfsBounty, newIpfsCallback);
       };
 
-      do_bounty();
+      var payFeaturedBounty = function() {
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/new_bounty.js]: use `const` ?

File: app/assets/v2/js/pages/new_bounty.js [hunk 2]
```diff
@@ -464,3 +490,11 @@ var check_balance_and_alert_user_if_not_enough = function(tokenAddress, amount)
     }
   });
 };
+
+let usdFeaturedPrice = $('.featured-price-usd').text();
+let ethFeaturedPrice;
+
+getAmountEstimate(usdFeaturedPrice, 'ETH', function(amountEstimate) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/new_bounty.js]: arrow function ?

File: app/assets/v2/js/shared.js [hunk 3]
```diff
@@ -1201,6 +1201,39 @@ function renderBountyRowsFromResults(results, renderForExplorer) {
   return html;
 }
 
+function saveAttestationData(result, cost_eth, to_address, type) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/shared.js]: could we change it to variable functions `const saveAttestationData = (result, cost_eth, to_address, type) => { }`  ?

File: app/assets/v2/js/shared.js [hunk 4]
```diff
@@ -1201,6 +1201,39 @@ function renderBountyRowsFromResults(results, renderForExplorer) {
...
+  var request_url = '/revenue/attestations/new';
+  var txid = result;
+  var data = {
+    'txid': txid,
+    'amount': cost_eth,
+    'network': document.web3network,
+    'from_address': web3.eth.coinbase,
+    'to_address': to_address,
+    'type': type
+  };
```
Comment (thelostone-mc) [inline app/assets/v2/js/shared.js]: update to let/const ?

File: app/assets/v2/js/shared.js [hunk 5]
```diff
@@ -1201,6 +1201,39 @@ function renderBountyRowsFromResults(results, renderForExplorer) {
...
+
+  $.post(request_url, data).then(function(result) {
+    _alert('Success ✅ Loading your purchase now.', 'success');
+  });
+}
+
+function renderFeaturedBountiesFromResults(results, renderForExplorer) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/shared.js]: variable function + arrow function

File: app/assets/v2/js/shared.js [hunk 6]
```diff
@@ -1201,6 +1201,39 @@ function renderBountyRowsFromResults(results, renderForExplorer) {
...
+  let html = '';
+  const tmpl = $.templates('#featured-card');
+
+  if (results.length === 0) {
+    return html;
+  }
+
+  for (var i = 0; i < results.length; i++) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/shared.js]: let

File: app/dashboard/templates/dashboard/featured_bounties.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{% load static i18n %}
+<div class="featured-bounties my-4" style="display: none;">
+    <div class="">
```
Comment (thelostone-mc) [inline app/dashboard/templates/dashboard/featured_bounties.html]: remove class :P

File: app/dashboard/templates/dashboard/featured_bounties.html [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+      <h3 class="featured-bounties__title">
+        Featured Bounties
```
Comment (thelostone-mc) [inline app/dashboard/templates/dashboard/featured_bounties.html]: wrap in trans

File: app/dashboard/templates/dashboard/featured_bounties.html [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+        <span class="status-open">
+          <i class="g-icon g-icon__dot-circle"></i>
+          Ready to work
```
Comment (thelostone-mc) [inline app/dashboard/templates/dashboard/featured_bounties.html]: wrap in trans

File: app/dashboard/templates/dashboard/featured_bounties.html [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+        </span>
+      </h3>
+    </div>
+    <div class="scroll-carousel" id="featured-card-container"></div>
+  </div>
```
Comment (thelostone-mc) [inline app/dashboard/templates/dashboard/featured_bounties.html]: indent fix needed

File: app/dashboard/templates/dashboard/index.html [hunk 11]
```diff
@@ -35,23 +37,27 @@
...
+            <div id="dashboard-title" class="hidden">
+              {% include 'shared/search_bar.html' %}
+            </div>
+            <div class="row loading justify-content-center align-items-center loading_img">
+              <img src="{% static "v2/images/loading_v2.gif" %}">
+            </div>
+            <div class="row nonefound" style="display:none;">
+                <img src="{% static "v2/images/shrug_bot.png" %}">
+                <h3>{% trans "No results found." %}</h3>
+                <p>{% trans "Please try another query" %}</p>
```
Comment (thelostone-mc) [inline app/dashboard/templates/dashboard/index.html]: indent fix for L55 to L57

File: app/dashboard/templates/dashboard/index.html [hunk 12]
```diff
@@ -65,12 +71,21 @@ <h3>{% trans "No results found." %}</h3>
...
+    {% include 'shared/featured_bounty_cards.html' %}
 
     {% include 'shared/analytics.html' %}
     {% include 'shared/footer_scripts.html' %}
     {% include 'shared/footer.html' %}
     {% include 'shared/messages.html' %}
+    <script src="{% static "v2/js/popper.min.js" %}"></script>
+    <script src="{% static "v2/js/bootstrap.min.js" %}"></script>
+    <script>
+    var bootstrapTooltip = $.fn.tooltip.noConflict()
```
Comment (thelostone-mc) [inline app/dashboard/templates/dashboard/index.html]: let?

File: app/dashboard/templates/shared/featured.html [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+              <span class="featured-price"><b class="featured-price-eth">ETH</b> (<span class="featured-price-usd">12</span> USD)</span>
+            </label>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+  <div class="col-md-5 text-right">
+    <img src="{% static 'v2/images/featured-example.jpg' %}" width="314" alt="featured bounty example" class="featured-example w-auto">
+  </div></co>
```
Comment (thelostone-mc) [inline app/dashboard/templates/shared/featured.html]: `</co>` ?

File: app/dashboard/templates/shared/featured.html [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+
```
Comment (thelostone-mc) [inline app/dashboard/templates/shared/featured.html]: remove extra empty lines?

**Comment by thelostone-mc** [comment]
left a few comments @octavioamu

**Comment by octavioamu** [comment]
Thanks @thelostone-mc I will address these

File: app/dashboard/templates/shared/featured.html (hunk 13 above)
Comment (octavioamu) [reply app/dashboard/templates/shared/featured.html]: no idea where this comes from lol

**Pull request merged**